DfE.FindInformationAcademiesTrusts/Services/Export/ExportService.cs

```diff
@@ -67,75 +83,226 @@ private static byte[] GenerateSpreadsheet(
-            {
-                academyDetails.EstablishmentName ?? string.Empty,
-                urn,
-                academyDetails.LocalAuthority ?? string.Empty,
-                academyDetails.TypeOfEstablishment ?? string.Empty,
-                academyDetails.UrbanRural ?? string.Empty,
-                ofstedData?.DateAcademyJoinedTrust.ToString(StringFormatConstants.ViewDate) ?? string.Empty,
-                previousRating.OverallEffectiveness.ToDisplayString() ?? string.Empty,
-                IsOfstedRatingBeforeOrAfterJoining(previousRating.OverallEffectiveness,
-                    ofstedData?.DateAcademyJoinedTrust ?? DateTime.MinValue, previousRating.InspectionDate),
-                previousRating.InspectionDate?.ToString(StringFormatConstants.ViewDate) ?? string.Empty,
-                currentRating.OverallEffectiveness.ToDisplayString() ?? string.Empty,
-                IsOfstedRatingBeforeOrAfterJoining(currentRating.OverallEffectiveness,
-                    ofstedData?.DateAcademyJoinedTrust ?? DateTime.MinValue, currentRating.InspectionDate),
-                currentRating.InspectionDate?.ToString(StringFormatConstants.ViewDate) ?? string.Empty,
-                pupilNumbersData?.PhaseOfEducation ?? string.Empty,
+            SetTextCell(worksheet, i + 4, 13, pupilNumbersData?.PhaseOfEducation ?? string.Empty);
+            SetTextCell(worksheet, i + 4, 14,
                 pupilNumbersData != null
                     ? $"{pupilNumbersData.AgeRange.Minimum} - {pupilNumbersData.AgeRange.Maximum}"
-                    : string.Empty,
-                pupilNumbersData?.NumberOfPupils?.ToString() ?? string.Empty,
-                pupilNumbersData?.SchoolCapacity?.ToString() ?? string.Empty,
-                percentageFull > 0 ? $"{percentageFull}%" : string.Empty,
+                    : string.Empty
+            );
+            SetTextCell(worksheet, i + 4, 15, pupilNumbersData?.NumberOfPupils?.ToString() ?? string.Empty);
+            SetTextCell(worksheet, i + 4, 16, pupilNumbersData?.SchoolCapacity?.ToString() ?? string.Empty);
+
+            SetTextCell(worksheet, i + 4, 17, percentageFull > 0 ? $"{percentageFull}%" : string.Empty);
+            SetTextCell(worksheet, i + 4, 18,
                 freeSchoolMealsData?.PercentageFreeSchoolMeals.HasValue == true
                     ? $"{freeSchoolMealsData.PercentageFreeSchoolMeals}%"
                     : string.Empty
-            };
+            );
+        }
+
+        worksheet.Columns().AdjustToContents();
+        return SaveWorkbookToByteArray(workbook);
+    }
+
+    private static void GenerateAcademiesRows(AcademyDetails[] academies, AcademyOfsted[] academiesOfstedRatings, AcademyPupilNumbers[] academiesPupilNumbers, AcademyFreeSchoolMeals[] academiesFreeSchoolMeals, IXLWorksheet worksheet, int i, out OfstedRating currentRating, out AcademyPupilNumbers? pupilNumbersData, out AcademyFreeSchoolMeals? freeSchoolMealsData, out float percentageFull)
```

Comment:
This method should only generate a single row, so it doesn't need to take in any of these arrays, just the single academy, ofsted rating etc.

Also this will mean that i is no longer used so you can pass the row value directly.

Also I don't understand why some of the cells are set in this method and some are set back in the main loop using the outputs from this method. Why not just set everything in this method and have no outputs?